android/src/main/java/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.java

```diff
@@ -31,18 +38,27 @@
 /**
  * PurchasesFlutterPlugin
  */
-public class PurchasesFlutterPlugin implements MethodCallHandler {
+public class PurchasesFlutterPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {
 
     private static final String PURCHASER_INFO_UPDATED = "Purchases-PurchaserInfoUpdated";
 
-    private final Registrar registrar;
-    private final MethodChannel channel;
+    // Only set registrar for v1 embedder.
+    private PluginRegistry.Registrar registrar;
+    // Only set activity for v2 embedder. Always access activity from getActivity() method.
+    private Context applicationContext;
+    private MethodChannel channel;
+    private Activity activity;
+
     private static final String PLATFORM_NAME = "flutter";
     private static final String PLUGIN_VERSION = "1.2.0-SNAPSHOT";
 
-    public PurchasesFlutterPlugin(Registrar registrar, MethodChannel channel) {
-        this.registrar = registrar;
-        this.channel = channel;
+    /**
+     * Plugin registration.
+     */
+    public static void registerWith(Registrar registrar) {
+        PurchasesFlutterPlugin instance = new PurchasesFlutterPlugin();
+        instance.onAttachedToEngine(registrar.messenger(), registrar.context());
+        instance.registrar = registrar;
         registrar.addViewDestroyListener(new PluginRegistry.ViewDestroyListener() {
```

Comment:
I am not sure we need this, but I don't think it would hurt to keep it since we already have it.